addons/mail/static/src/models/channel_member_list_view.js

```diff
@@ -1,32 +1,60 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { one } from '@mail/model/model_field';
-import { clear, replace } from '@mail/model/model_field_command';
+import { attr, one } from '@mail/model/model_field';
+import { clear, replace, insert} from '@mail/model/model_field_command';
 
 registerModel({
     name: 'ChannelMemberListView',
     identifyingFields: [['chatWindowOwner', 'threadViewOwner']],
+    lifecycleHooks: {
+        _created() {
+            this.fetchChannelMembers();
+        }
+    },
     recordMethods: {
+        /**
+         * Handles click on the "load more members" button.
+         */
+        async onClickLoadMoreMembers() {
+            this.fetchChannelMembers();
+        },
+        async fetchChannelMembers() {
+            const channelMembersList = await this.messaging.rpc({
+                model: 'mail.channel',
+                method: 'load_more_members',
+                args: [[this.channel.id]],
+                kwargs: {
+                    'known_member_ids': this.channel.channelMembers.map(channelMember => channelMember.partner.id),
+                },
+            });
+            this.channel.update({ memberCount: channelMembersList.count });
+            for (const partner of channelMembersList.partners) {
+                const relation = { channel: replace(this.channel) };
+                relation.partner = insert(partner);
+                this.channel.update({ channelMembers: insert(relation) });
+            }
+        },
```

Comment:
It's much better to just do the RPC, and receive response as longpolling notification to insert data.
This scales much better with code changes/improvements/increase of feature scope/etc.

Reply:
I don't know. This a preparation PR to introduce guest in the member list. I don't really want to introduce new behavior/way to handle data transfer.